### PR TITLE
getlantern/lantern#2162 Restricted outbound access to just ports 80, 443...

### PR DESCRIPTION
**Pull request by oxtoacart**
..., 8080 and 8443

### CODE REVIEW

File: src/github.com/getlantern/fronted/server.go [hunk 1]
```diff
@@ -169,27 +174,63 @@ func (server *Server) Serve(l net.Listener) error {
...
 }
 
+func (server *Server) checkForLocalAddress(addr string) error {
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return fmt.Errorf("Unable to split host and port for %v: %v", addr, err)
+	}
+
+	ipAddr, err := net.ResolveIPAddr("ip", host)
+	if err != nil || len(ipAddr.IP) == 0 {
```
Comment (oxtoacart) [inline src/github.com/getlantern/fronted/server.go]: This was a drive-by fix.  I noticed that if passing in something like ":0", ResolveIPAddr would not return an error, but line 206 would fail because IP is empty.

Comment (myleshorton) [inline src/github.com/getlantern/fronted/server.go]: Cool. Maybe a better name would be checkForGlobalUnicast?

File: src/github.com/getlantern/fronted/server.go [hunk 2]
```diff
@@ -169,27 +174,63 @@ func (server *Server) Serve(l net.Listener) error {
...
+
+func (server *Server) checkForDisallowedPort(addr string) error {
+	_, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		return fmt.Errorf("Unable to split host and port for %v: %v", addr, err)
+	}
+
+	portAllowed := false
+	for _, allowed := range server.AllowedPorts {
+		if port == strconv.Itoa(allowed) {
```
Comment (myleshorton) [inline src/github.com/getlantern/fronted/server.go]: Maybe slightly better to do this outside of the loop, as in -- not a biggie of course, but since every request hits this:

```go
port := strconv.Atoi(portString)
for _, allowed := range server.AllowedPorts {
		if port == allowed {
```

File: src/github.com/getlantern/fronted/server.go (hunk 1 above)
Comment (oxtoacart) [inline src/github.com/getlantern/fronted/server.go]: Renamed to checkForNonGlobalDestination.

File: src/github.com/getlantern/fronted/server.go (hunk 2 above)
Comment (oxtoacart) [inline src/github.com/getlantern/fronted/server.go]: Fixed.

**Comment by myleshorton** [comment]
LGTM.
